src/csharp/Microsoft.Spark/Utils/AssemblyLoader.cs

```diff
@@ -169,5 +182,16 @@ private static bool TryLoadAssembly(string assemblyFileName, ref Assembly assemb
 
             return false;
         }
+
+        /// <summary>
+        /// Normalizes the assemblyName by removing characters known to cause
+        /// issues. This is useful in situations where the assemblyName is
+        /// automatically generated, ie the Roslyn compiler used in the REPL
+        /// generates an assembly name that contains * and #.
+        /// </summary>
+        /// <param name="assemblyName">Assembly name</param>
+        /// <returns>Normalized assembly name</returns>
+        private static string NormalizeAssemblyName(string assemblyName) =>
```

Comment:
do we need to normalize the assembly name even if assembly file name is not used?